codecoverage.py

```diff
@@ -91,10 +91,15 @@ def generate_info(grcov_path):
         else:
             ordered_files.append("ccov-artifacts/" + fname)
 
+    # Assume we're on a one-click loaner.
+    mod_env = os.environ.copy()
+    if os.path.isdir('/home/worker/workspace/build/src/gcc/bin'):
```

Comment:
In my experience, when a user first gets access to the one-click loaner shell, the build job is still running in the background, so this directory won't exist or will be empty for at least 20 minutes. You could add some error handling here to help them understand that, in case they run this script too early.